src/components/navigation/document-tab-panel.tsx

```diff
@@ -26,15 +26,32 @@ interface IState {
 
 const kNavItemScale = 0.11;
 
+export interface ISubTabSpec {
+  label: string;
+  sections: NavTabSectionSpec[];
+}
+
 @inject("stores")
 @observer
 export class DocumentTabPanel extends BaseComponent<IProps, IState> {
 
+  subTabs: ISubTabSpec[] = [];
+
   constructor(props: IProps) {
     super(props);
     this.state = {
       tabIndex: 0
     };
+
+    props.tabSpec.sections?.forEach(section => {
+      const found = this.subTabs.findIndex(tab => tab.label === section.title);
+      if (found >= 0) {
+        this.subTabs[found].sections.push(section);
+      }
+      else {
+        this.subTabs.push({ label: section.title, sections: [section] });
+      }
+    });
```

Comment:
Probably a clever way to collapse this into a map.  But I'm more curious as to if `subTabs` should be in state rather than a class variable.

Reply:
The important point is that this collapses sections with the same title, which is how we specify that problem and personal documents should be combined into a single section in the `app-config.json`.